Modules/mmapmodule.c

```diff
@@ -264,7 +375,44 @@ mmap_read_byte_method(mmap_object *self,
         PyErr_SetString(PyExc_ValueError, "read byte out of range");
         return NULL;
     }
-    return PyLong_FromLong((unsigned char)self->data[self->pos++]);
+    char dest;
+    if (safe_byte_copy(&dest, self->data + self->pos) < 0) {
+        return NULL;
+    }
+    else {
+        self->pos++;
+        return PyLong_FromLong(dest);
```

Comment:
```suggestion
        return PyLong_FromLong((unsigned char)dest);
```

For the failing test.